scripts/Windows/qt-installer-windows.qs

```diff
@@ -0,0 +1,292 @@
+    "qt.qt5.598.qtpurchasing.win32_msvc2015",
+    "qt.qt5.598.qtpurchasing.win64_msvc2017_64",
+    "qt.qt5.598.qtremoteobjects",
+    "qt.qt5.598.qtremoteobjects.win32_mingw53",
+    "qt.qt5.598.qtremoteobjects.win32_msvc2015",
+    "qt.qt5.598.qtremoteobjects.win64_msvc2017_64",
+    "qt.qt5.598.qtscript",
+    "qt.qt5.598.qtscript.win32_mingw53",
+    "qt.qt5.598.qtscript.win32_msvc2015",
+    "qt.qt5.598.qtscript.win64_msvc2017_64",
+    "qt.qt5.598.qtspeech",
+    "qt.qt5.598.qtspeech.win32_mingw53",
+    "qt.qt5.598.qtspeech.win32_msvc2015",
+    "qt.qt5.598.qtspeech.win64_msvc2017_64",
+    "qt.qt5.598.qtvirtualkeyboard",
+    "qt.qt5.598.qtvirtualkeyboard.win32_mingw53",
+    "qt.qt5.598.qtvirtualkeyboard.win32_msvc2015",
+    "qt.qt5.598.qtvirtualkeyboard.win64_msvc2017_64",
+    "qt.qt5.598.qtwebengine",
+    "qt.qt5.598.qtwebengine.win32_msvc2015",
+    "qt.qt5.598.qtwebengine.win64_msvc2017_64",
+
+    // Tools
+    "qt.tools.win32_mingw530",
+    "qt.tools.win32_mingw730",
+    "qt.tools.win64_mingw730",
+    "qt.tools.ifw.20",
+    "qt.tools.ifw.30",
+    "qt.tools.ifw.31"
+];
+
+function Controller() {
+    installer.autoRejectMessageBoxes();
+    installer.installationFinished.connect(function() {
+        gui.clickButton(buttons.NextButton);
+    })
+}
+
+Controller.prototype.WelcomePageCallback = function() {
+    // click delay here because the next button is initially disabled for ~1 second
+    gui.clickButton(buttons.NextButton, 3000);
+}
+
+Controller.prototype.CredentialsPageCallback = function() {
+    gui.clickButton(buttons.NextButton);
+}
+
+Controller.prototype.IntroductionPageCallback = function() {
+    gui.clickButton(buttons.NextButton);
+}
+
+Controller.prototype.DynamicTelemetryPluginFormCallback = function() {
+    gui.currentPageWidget().TelemetryPluginForm.statisticGroupBox.disableStatisticRadioButton.setChecked(true);
+    gui.clickButton(buttons.NextButton);
+
+    //for(var key in widget.TelemetryPluginForm.statisticGroupBox){
+    //    console.log(key);
+    //}
+}
+
+Controller.prototype.TargetDirectoryPageCallback = function()
+{
+    // Keep default at "C:\Qt".
+    //gui.currentPageWidget().TargetDirectoryLineEdit.setText("E:\\Qt");
+    //gui.currentPageWidget().TargetDirectoryLineEdit.setText(installer.value("HomeDir") + "/Qt");
+    gui.clickButton(buttons.NextButton);
+}
+
+Controller.prototype.ComponentSelectionPageCallback = function() {
+
+    // https://doc-snapshots.qt.io/qtifw-3.1/noninteractive.html
+    var page = gui.pageWidgetByObjectName("ComponentSelectionPage");
+
+    // if CategoryGroupBox is visible, check one of the checkboxes
+    // and click fetch button before selecting any components
+    var groupBox = gui.findChild(page, "CategoryGroupBox");
+    if (groupBox) {
+        console.log("groupBox found");
+        // findChild second argument is the display name of the checkbox
+        var checkBox = gui.findChild(page, "Archive");
```

Comment:
I appreciate this script! It helped me figure out some issues with my own script.

Based on my testing, it seems that selecting build chains for `5.13.1` will silently fail unless `"Latest releases"` is also selected. Here's my `ComponentSelectionPageCallback`, which seems to be working as expected for my case.

```javascript
Controller.prototype.ComponentSelectionPageCallback = function() {
    var page = gui.pageWidgetByObjectName("ComponentSelectionPage");

    var checkBox = gui.findChild(page, "Latest releases");
    var fetchButton = gui.findChild(page, "FetchCategoryButton");

    checkBox.click();
    fetchButton.click();
    
    var widget = gui.currentPageWidget();

    widget.deselectAll();
    widget.selectComponent("qt.qt5.5131.win64_mingw73");
    widget.selectComponent("qt.tools.win64_mingw730");

    gui.clickButton(buttons.NextButton);
}
```

Reply:
You are welcome! Took some time though to understand the mechanics of Qt installer UI :)

Reply:
I worded it poorly, but my comment was partly to point out a potential error with the script as it is right now. Just want to make sure you see it:

As it is right now, your script will not install any components from Qt 5.13.1. Qt 5.13 is now in its own category, `"Latest releases"`, which is not selected by default. The worst part about this error is that the installer will run through "successfully", silently failing when asked to select a nonexistent component.

Thanks again for writing this, and best of luck maintaining it! I know how much of a headache it can be 😄